install_api.sh

```diff
@@ -444,6 +445,33 @@ setup_api() {
     else
         print "\nSkipping service installation."
     fi
+
+    # enable HTTPS
+    enable_https
+
+}
+
+enable_https() {
+    https=$(get_configuration_value "https")
+    if [ "X${https}" == "Xyes" ]; then
+        # generate a self-signed certificate if it does not exist
+        if ! ([ -f "$API_PATH/configuration/ssl/server.key" ] || [ -f "$API_PATH/configuration/ssl/server.crt" ]); then
+            default_pass="wazuh"
+            subject="/C=XX/ST=XX/L=XX/O=XX/OU=XX/CN=XX"
+            if command -v openssl > /dev/null 2>&1; then
+                exec_cmd_bash "cd $API_PATH/configuration/ssl && openssl genrsa -des3 -out server.key -passout pass:${default_pass} 2048 && cp server.key server.key.org && openssl rsa -in server.key.org -out server.key -passin pass:${default_pass}"
```

Comment:
This function can be simplified by using the script `configure_api.sh`. This script accepts a `preloaded_vars.conf` file to automate the execution of that script and you can set up the user, the port, the https, etc. 

So I think that you can use that script to set up just the https.

Reply:
In addition, for security reasons, it's better to create an RSA key with a length of 4096 bits rather than 2048 bits.